pipert2/core/wrappers/api_wrapper.py

```diff
@@ -0,0 +1,84 @@
+import time
+
+import flask
+from flask import Flask, request
+from pipert2 import Pipe, SourceRoutine, Data, DestinationRoutine
+from flask import Response
+from multiprocessing import Process
+from pipert2.utils.consts import START_EVENT_NAME, STOP_EVENT_NAME, KILL_EVENT_NAME
+
+
+class APIWrapper:
+    def __init__(self, host: str, port: int, pipe: Pipe):
+        """Api wrapper for notify events through HTTP.
+
+        Args:
+            host: The host the API will run on.
+            port: The port the API will run in.
+            pipe: The pipe to notify event through it.
+        """
+
+        self.notify_callback = pipe.get_event_notify()
+
+        self.app = Flask(__name__)
+        self.app.add_url_rule("/start", "start", self.start)
+        self.app.add_url_rule("/pause", "pause", self.pause)
+        self.app.add_url_rule("/kill", "kill", self.kill)
+        self.app.add_url_rule("/execute", "execute", self.execute)
```

Comment:
I think its better to see the url routes above each method, 
but i guess you tried and it caused bugs with the self variable ?

Reply:
Yep, when using classes the 'above the function' didn't work and this is the alternative.